drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -281,6 +281,68 @@ def self.generate_enrollment_id
     generate_uuid
   end
 
+  # Data Collection Features that are enabled for this enrollment, thru its project (e.g. Current Living Situation)
+  #
+  # Is it enabled?
+  #   If ANY data exists for it in this enrollment, even if no active instances exist, then yes.
+  #   (This supports both inactive instances and migrated-in data, as well as context changes such as change in HoH)
+  #
+  # Who is data collected about?
+  #   Choose the "best" instance – IE the one that would actually be selected
+  #   when creating a new record – and return the data_collected_about on it.
+  #
+  # Returns an OpenStruct that is resolved by the DataCollectionFeature GQL type.
+  def data_collection_features
+    # Create OpenStruct for each enabled feature
+    Hmis::Form::Definition::DATA_COLLECTION_FEATURE_ROLES.map do |role|
+      instance_scope = Hmis::Form::Instance.with_role(role).active.published
+      # Service instances must specify a service type or category.
+      instance_scope = instance_scope.for_services if role == :SERVICE
+
+      # Choose the "best" instance, i.e. the one that would actually be selected when recording new data.
+      # We need to do this so that we can accurately set "data collected about" based on the most applicable form.
+      # This detect_best_instance_for_project logic is also called from the query to resolve the actual form definition
+      best_instance = instance_scope.detect_best_instance_for_project(project: project)
+
+      has_any_data = case role
+      when :CURRENT_LIVING_SITUATION
+        current_living_situations.exists?
+      when :SERVICE
+        custom_services.exists? || services.exists?
+      when :CE_EVENT
+        events.exists?
+      when :CE_ASSESSMENT
+        assessments.exists?
+      when :CASE_NOTE
+        custom_case_notes.exists?
+      when :REFERRAL
+        false # Referrals are resolved only on projects, not enrollments
+      when :REFERRAL_REQUEST
+        false # Referrals are resolved only on projects, not enrollments
```

Comment:
This is a little confusing to me. It seems like we DO resolve them on Enrollments if we find a `best_instance`, since they are included here.  If they _shouldn't_ be resolve on Enrollments (which I agree with), then we should exclude them from the list (`Hmis::Form::Definition::DATA_COLLECTION_FEATURE_ROLES.excluding(:REFERRAL, :REFERRAL_REQUEST, :EXTERNAL_FORM`).

Reply:
Yeah that's true. I think by "resolved" I meant to say "resolved by the frontend" but I should've said "*used* by the frontend" since it's resolved either way.

I'll make the change you're suggesting.